library/x509_crt.c

```diff
@@ -1435,32 +1569,195 @@ int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path )
 }
 #endif /* MBEDTLS_FS_IO */
 
+/*
+ * OtherName ::= SEQUENCE {
+ *      type-id    OBJECT IDENTIFIER,
+ *      value      [0] EXPLICIT ANY DEFINED BY type-id }
+ *
+ * HardwareModuleName ::= SEQUENCE {
+ *                           hwType OBJECT IDENTIFIER,
+ *                           hwSerialNum OCTET STRING }
+ *
+ * NOTE: we currently only parse and use otherName of type HwModuleName,
+ * as defined in RFC 4108.
+ */
+static int x509_get_other_name( const mbedtls_x509_sequence *subject_alt_name,
```

Comment:
This function doesn't use the linked list structure of `mbedtls_x509_sequence` but builds on `subject_alt_name->buf` only; is there a reason not to use a `mbedtls_x509_buf*` argument here?

Reply:
I agree. I'll change